src/panoramic/cli/cli.py

```diff
@@ -211,3 +211,188 @@ def scaffold(target_dataset: str, yes: bool):
     )
 
     scaffold_missing_fields_command(target_dataset=target_dataset, yes=yes)
+
+
+@cli.group()
+def connection():
+    """Connection subcommand for managing connections.
+
+    \b
+    All connections are stored in ~/.pano/config file.
+    You can edit this file either manually or using provided commands.
+
+    Expected YAML structure:
+
+    \b
+    name:
+      type: postgres
+      user: my_user
+      password: <password>
+      host: localhost
+      port: 5432
+      database: my_db
+
+    Which effectively results in a connection string similar to: "postgres://my_user@<password>@localhost:5432/my_db".
+
+    \b
+    name: The name of the connection that will be used as reference to specify which connection to use.
+    type: The type of data warehouse you are connecting to.
+    user: The username used for database connection.
+    password: The password used for database connection.
+    host: The hostname (server) used for database connection.
+    port: The port used for database connection.
+    database: The database used for database connection.
+    schema: The schema to build models into by default. Can be overridden by custom models.
+    warehouse: The warehouse to use when building models. (Used by Snowflake)
+    account: The account used for database connection. (Used by Snowflake)
+    project: The project used for database connection. (Used by BigQuery)
+    key_file: Keyfile path to Service Account JSON. (Used by BigQuery)
+    """
+    pass
+
+
+@connection.command()
+@click.argument('name', type=str)
+@click.option('--type', type=str, help='Type of the database. E.g. "postgres", "snowflake".')
+@click.option('--user', default='', type=str, help='Connection username.')
+@click.option('--password', default='', type=str, help='Connection password.')
+@click.option('--password-stdin', default=False, is_flag=True, help='Read connection password from standard input.')
+@click.option('--host', default='', type=str, help='Connection hostname.')
+@click.option('--port', default=0, type=int, help='Connection port.')
+@click.option('--database', default='', type=str, help='Connection database name.')
+@click.option('--schema', default='', type=str, help='Connection schema name.')
+@click.option('--warehouse', default='', type=str, help='Connection warehouse name. (Used by Snowflake)')
+@click.option('--account', default='', type=str, help='Connection account name. (Used by Snowflake)')
+@click.option('--project', default='', type=str, help='Connection project name. (Used by BigQuery)')
```

Comment:
Not a big fan of using `''` to mean not set - can we use `None` instead?

Reply:
Replaced for None.